source/SemVerAnalyzer/Nuget/NugetClient.cs

```diff
@@ -21,7 +25,32 @@ public async Task<byte[]> GetAssemblyBytesFromPackage(string packageName, string
 			try
 			{
 				using var client = new HttpClient();
-				using var request = new HttpRequestMessage(HttpMethod.Get, Path.Join(_config.RepositoryUrl, packageName));
+				
+				using var feedRequest = new HttpRequestMessage(HttpMethod.Get, Path.Combine(_config.RepositoryUrl, "v3", "index.json"));
+				using var feedResponse = await client.SendAsync(feedRequest);
+				if (!feedResponse.IsSuccessStatusCode)
+				{
+					var content = await feedResponse.Content.ReadAsStringAsync();
+					comments.Add($"Error retrieving Nuget feed:\n{content}");
+					return null;
+				}
+
+				var feed = JsonSerializer.Deserialize<NugetPackageFeed>(await feedResponse.Content.ReadAsStringAsync());
+				var packageBaseAddress = feed.Resources.Single(r => r.Type.StartsWith("PackageBaseAddress")).Id;
+				
+				using var versionRequest = new HttpRequestMessage(HttpMethod.Get, Path.Combine(packageBaseAddress, packageName, "index.json"));
+				using var versionResponse = await client.SendAsync(versionRequest);
+				if (!versionResponse.IsSuccessStatusCode)
+				{
+					var content = await versionResponse.Content.ReadAsStringAsync();
+					comments.Add($"Error retrieving package versions:\n{content}");
+					return null;
+				}
```

Comment:
As this code group appears three times with essentially the same statements, I'd really prefer it to be extracted into a method

Reply:
Agreed, I'll make that change 🧹